### ISSUE TITLE
Store snippets in workspace

### ISSUE THREAD

**Issue by efficacy** [Zettlr/Zettlr#3775]
### Checklist

- [X] Zettlr does not possess this feature (if it does, but the feature doesn't work out for you, propose an enhancement instead.)
- [X] Zettlr's current features cannot be used to approximate this behaviour (otherwise, propose an enhancement instead.)
- [X] I have searched the documentation to make sure I did not overlook it.
- [X] I have searched both open and closed issues to see if someone has already proposed this.

### Feature Proposal

Store snippets in the workspace instead of, or in addition to, the current method and location.

- The current single global store must contain all snippets for every workspace, but snippets are often project- or content- specific.
- Forcing all snippets into one store can cause name clashes, or at the very least require more disambiguating keypresses to activate.
- Snippets are not available when a workspace on a shared drive is opened on a different machine.
- Transferring snippets to a new machine is fiddly and requires technical knowledge about how Zettlr stores its configurations

### Example Workflow

This catches me out every time I open Zettlr on a new machine. I load up my workspace, read a few things, then attempt to create a new document. My fingers go to the sequence to insert one of my standard frontmatter snippets and .... nothing happens. I need to go back to a machine which I have already set up, dig around untl I find where the snippets are stored, and manually copy them and upload them to somewhere I can "see" from the new machine.

Snippets are currently stored in the user preferences hidden away on each machine, not in the workspace which lives on a shared drive and is available everywhere.

### Caveats

Completely changing the way snippets are stored could upset users who would need to find or manually re-enter snippets for the new system, so I suggest a dual system in the first instance, which either merges both workspace and global snippets, or automatically migrates global snippets into a workspace if it has none.

### Do you wish to attempt implementing this yourself?

No

### Platforms

- [X] Windows
- [X] macOS
- [X] Linux

**Comment by nathanlesage** [comment]
These sound like two separate issues: One about an easier way to access your snippets and one to hold additional snippets in directories. Which one should we focus on here?

**Comment by efficacy** [comment]
I can see that.

Personally, I don't really care whether the snippets in the workspace replace or extend ones stored in the machine. What I need is to be able to get up and running with Zettlr on a new machine, even if I don't have access to the previous one.

- I store my workspaces on a shared drive (in my case Google Drive) so I can connect from wherever I am
- Although I have my favourite UI settings, they are easy enough to change, or I can live with the defaults
- But snippets are much more complex and can't easily be re-created.

If you can think of a different way to access snippets without needing an explicit action on the donor machine, then I'd probably be happy with that,. It just seems to me that storing snippets in a workspace seems a stronger feature, and also offers the other advantages I mentioned.

Does that make sense?

**Comment by nathanlesage** [comment]
Yes, then let me propose the following: Since there's a ton of good data now being stored for Zettlr, one of my relative mid-term aims would be to offer a super simple import/export mechanism for all of that.

For example, one might be able to "export" all custom user-data (e.g. just in a zip-file), and then on "import" choose which data one may want to re-import or something like that.

**Comment by efficacy** [comment]
That's certainly better than what we have now, and would cope with a lot of cases, so by all means implement it!

It still doesn't help if you find yourself separated from the original machine, though.
For example, if that machine has failed, or run out of battery, or is simply a long distance away.

What would help in such cases is::

- storing the settings in the workspace as above  _or_
- some sort of automatic backup export of settings to a shared place - either on a regular basis, or when anything is changed or added

I could live with either of these. Even though I prefer the idea of per-workspace settings (for reasons given above), I can see how import/export is a simpler step from where we are now.

**Comment by kunashu** [comment]
Hello this seems like a really good idea. I was wondering if there has been any updates? For context I like to share my knowledge base throughout several computers using git for managing my note's cloud storage. One thing that would be very useful is to include the snippets in the git repository that way I can easily share them across all my computers.
